core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp

```diff
@@ -463,61 +445,30 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
 #else
     m_instance = arg_policy.space().impl_internal_space_instance();
 #endif
-    /*static_assert( std::is_same< typename ViewType::memory_space
-                                    , Kokkos::HostSpace >::value
-      , "Reduction result on Kokkos::OpenMP must be a Kokkos::View in HostSpace"
-      );*/
-  }
-
-  inline ParallelReduce(const FunctorType& arg_functor, Policy arg_policy,
-                        const ReducerType& reducer)
-      : m_instance(nullptr),
-        m_functor(arg_functor),
-        m_policy(arg_policy),
-        m_reducer(reducer),
-        m_result_ptr(reducer.view().data()) {
-#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_3
-    if (t_openmp_instance) {
-      m_instance = t_openmp_instance;
-    } else {
-      m_instance = arg_policy.space().impl_internal_space_instance();
-    }
-#else
-    m_instance = arg_policy.space().impl_internal_space_instance();
-#endif
-    /*static_assert( std::is_same< typename ViewType::memory_space
-                                    , Kokkos::HostSpace >::value
-      , "Reduction result on Kokkos::OpenMP must be a Kokkos::View in HostSpace"
-      );*/
+    static_assert(
+        std::is_same<typename ViewType::memory_space, Kokkos::HostSpace>::value,
+        "Reduction result on Kokkos::OpenMP must be a Kokkos::View in "
+        "HostSpace");
```

Comment:
I don't think that's correct. Why can't I use OpenMP with a view that uses pinned memory or  Cuda UVM ?

Reply:
Should the assertion be loosened to the space being host accessible?

Reply:
I agree. Fixed.